openverse_catalog/dags/providers/provider_api_scripts/museum_victoria.py

```diff
@@ -1,203 +1,175 @@
-                data = response_json
-                break
-        except Exception:
-            data = None
-    return data
-
-
-def _handle_batch_objects(objects, landing_page=LANDING_PAGE):
-    image_count = 0
-    for obj in objects:
-        object_id = obj.get("id")
-        if object_id in RECORDS_IDS:
-            continue
-        RECORDS_IDS.append(object_id)
-        foreign_landing_url = landing_page + object_id
-        media_data = obj.get("media")
-        if media_data is None:
-            continue
-        image_data = _get_media_info(media_data)
-        if len(image_data) == 0:
-            continue
-        meta_data = _get_metadata(obj)
-        title = obj.get("displayTitle")
-        for img in image_data:
-            license_info = get_license_info(license_url=img.get("license_url"))
-            image_count = image_store.add_item(
-                foreign_identifier=img.get("image_id"),
-                foreign_landing_url=foreign_landing_url,
-                image_url=img.get("image_url"),
-                height=img.get("height"),
-                width=img.get("width"),
-                license_info=license_info,
-                title=title,
-                creator=img.get("creators"),
-                meta_data=meta_data,
-            )
-    return image_count
-
-
-def _get_media_info(media_data):
-    image_data = []
-    for media in media_data:
-        media_type = media.get("type")
-        if media_type == "image":
-            image_id = media.get("id")
-            image_url, height, width = _get_image_data(media)
-            license_url = _get_license_url(media)
-            if image_url is None or image_id is None or license_url is None:
+class ImageDetails(TypedDict, total=False):
+    foreign_identifier: str
+    image_url: str
+    license_info: LicenseInfo
+    foreign_landing_url: str
+    title: str
+    meta_data: Dict
+    height: int | None
+    width: int | None
+    creator: str | None
+
+
+class VictoriaDataIngester(ProviderDataIngester):
+    providers = {"image": prov.VICTORIA_DEFAULT_PROVIDER}
+    endpoint = "https://collections.museumsvictoria.com.au/api/search"
+    headers = {"User-Agent": prov.UA_STRING, "Accept": "application/json"}
+    batch_limit = 100
+    delay = 5
+    # This set is used to prevent duplicate images of the same items
+    RECORDS_IDS = set()
+    LANDING_PAGE = "https://collections.museumsvictoria.com.au/"
+    LICENSE_LIST = [
+        "public domain",
+        "cc by",
+        "cc by-nc",
+        "cc by-nc-sa",
+        "cc by-nc-nd",
+        "cc by-sa",
+    ]
+
+    def ingest_records(self, **kwargs):
+        for license_ in VictoriaDataIngester.LICENSE_LIST:
+            super().ingest_records(license_=license_)
+
+    def get_batch_data(self, response_json):
+        if response_json:
+            return response_json
+        return None
+
+    def get_next_query_params(self, old_query_params: Optional[Dict], **kwargs) -> Dict:
```

Comment:
Note that this will soon be `prev_query_params` so it might be good to preemptively change that here: https://github.com/WordPress/openverse-catalog/pull/579#discussion_r919333596